### PR TITLE
Implement lineDraw

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Heavily based off of the OpenRCT2 version

Not tested or verified yet.

### CODE REVIEW

File: src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp [hunk 1]
```diff
@@ -1537,17 +1537,129 @@ namespace OpenLoco::Drawing
...
+
+            if ((left > (rt.x + rt.width)) && (right > (rt.x + rt.width)))
+            {
+                return;
+            }
+
+            if ((top > (rt.y + rt.height)) && (bottom > (rt.y + rt.height)))
+            {
+                return;
+            }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp:1608]: I think we have a helper function for this in `Rect::intersects`. We should probably change the parameter to a `Rect`, adding a compat function for the old signature. Thoughts?

**Review by AaronVanGeffen — COMMENTED** [review]
Comment aside, charts appear to be drawn just fine.

File: src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp:1608]: ah good point.

Comment (duncanspumpkin) [reply src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp]: I decided that giving a Rect as input to a line draw function would be weird. I've gone for two Ui::Points a and b. I then construct a Rect to do the intersect.

**Review by AaronVanGeffen — APPROVED** [review]
Haven't tested the latest changes, but the code looks much nicer to me.

**Pull request merged**